pkg/runtime/wfengine/wfengine.go

```diff
@@ -217,19 +187,27 @@ func (wfe *WorkflowEngine) Start(ctx context.Context) (err error) {
 		return errors.New("gRPC executor is not yet configured")
 	}
 
-	for actorType, actor := range wfe.InternalActors() {
+	for actorType, actor := range wfe.GetInternalActorsMap() {
 		err = wfe.actorRuntime.RegisterInternalActor(ctx, actorType, actor, time.Minute*1)
 		if err != nil {
 			return fmt.Errorf("failed to register workflow actor %s: %w", actorType, err)
 		}
 	}
 
-	// TODO: Determine whether a more dynamic parallelism configuration is necessary.
-	parallelismOpts := backend.WithMaxParallelism(100)
-
-	orchestrationWorker := backend.NewOrchestrationWorker(wfe.backend, wfe.executor, wfBackendLogger, parallelismOpts)
-	activityWorker := backend.NewActivityTaskWorker(wfe.backend, wfe.executor, wfBackendLogger, parallelismOpts)
+	// There are separate "workers" for executing orchestrations (workflows) and activities
+	orchestrationWorker := backend.NewOrchestrationWorker(
+		wfe.backend,
+		wfe.executor,
+		wfBackendLogger,
+		backend.WithMaxParallelism(wfe.spec.MaxConcurrentWorkflows))
```

Comment:
This is where we actually reference the new max concurrency configuration settings. Previously these values were hardcoded to 100 (see the deleted lines above here).